dogs_labeling/frontend/src/httpServices.ts

```diff
@@ -0,0 +1,48 @@
+import { getSessionId, setSessionId } from "./appState";
+import { myResponse, fetchParams } from "./models";
+//should these fetching functions be more generic?
+//receive parameters for method and url maybe
+export async function getPicsBackend(params: fetchParams) { //bad cast params as any 
+    let response = await fetch('http://localhost:8080/preferences?' + new URLSearchParams(params as any), {
+        method: "GET",
+        headers: { 'Content-Type': 'application/json' },
+    });
+    let responseJSON: myResponse = await response.json();
+    return responseJSON;
+}
+
+export async function postPreferences(body: any) {
+    let response = await fetch('http://localhost:8080/preferences', {
```

Comment:
```suggestion
    const baseUrl = 'http://localhost:8080/preferences';
    let response = await fetch(baseUrl, {
```

If for example the port your app is running on changes then you would have to go through all the instances where you use the hardcoded url and change it. In general, when you're using a value multiple times, think if would be useful to extract it as a const.

Reply:
followed your advice and i also created a generic fetch function
https://github.com/CatalinB7/dogs-labeling/pull/2/files#diff-b0abc765a3b7c7158a09dc4d11771c0cc58bf9a6d6f158c2791d2ad656fbb37fR4-R19